sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -175,6 +175,64 @@ object GpuCast extends Arm {
       cv.stringReplaceWithBackrefs(rule.search, rule.replace)
     })
   }
+
+  def sanitizeStringToFloat(input: ColumnVector): ColumnVector = {
+
+    // This regex gets applied after the transformation to normalize use of Inf and is
+    // just strict enough to filter out known edge cases that would result in incorrect
+    // values. We further filter out invalid values using the cuDF is_float method.
+    val VALID_FLOAT_REGEX =
+      "^" +                         // start of line
+      "[+\\-]?" +                   // optional + or - at start of string
+      "(" +
+        "(" +
+          "(" +
+            "([0-9]+)|" +           // digits, OR
+            "([0-9]*\\.[0-9]+)|" +  // decimal with optional leading and mandatory trailing, OR
+            "([0-9]+\\.[0-9]*)" +   // decimal with mandatory leading and optional trailing
+          ")" +
+          "([eE][+\\-]?[0-9]+)?" +  // exponent
+          "([fFdD])?" +             // floating-point designator
+        ")" +
+        "|(Inf)" +                  // Infinity
+        "|([nN][aA][nN])" +         // NaN
+      ")" +
+      "$"                           // end of line
+
+    withResource(input.lstrip()) { stripped =>
+      withResource(GpuScalar.from(null, DataTypes.StringType)) { nullString =>
+        // filter out strings containing breaking whitespace
+        val withoutWhitespace = withResource(stripped.containsRe("(\r|\n)")) {
+            hasVerticalWhitespace =>
+          hasVerticalWhitespace.ifElse(nullString, stripped)
+        }
+        // replace all possible versions of infinity with Inf
+        withResource(withoutWhitespace) { _ =>
+          withResource(withoutWhitespace.stringReplaceWithBackrefs(
+            "([iI][nN][fF])([iI][nN][iI][tT][yY])?", "Inf")) { inf =>
+            // replace +Inf with Inf
+            withResource(inf.stringReplaceWithBackrefs("([+])(Inf)", "Inf")) {
```

Comment:
Why is this using a regex with backrefs?  I don't see any need for backrefs since we aren't using any captured groups in the replacement.  Isn't this simply a standard replace of `+Inf` with `Inf`?  That would be much faster.